parachain/pallets/bridge/src/lib.rs

```diff
@@ -65,28 +75,20 @@ decl_module! {
 
 		fn deposit_event() = default;
 
-		/// Submit `message` for dispatch to a target application identified by `app_id`.
 		#[weight = 0]
-		pub fn submit(origin, app_id: AppId, message: Message) -> DispatchResult {
-			let who = ensure_signed(origin)?;
-
-			// TODO: move replay protection here
+		pub fn submit(origin, channel_id: ChannelId, message: Message) -> DispatchResult {
+			let relayer = ensure_signed(origin)?;
 
-			T::Verifier::verify(who, app_id, &message)?;
-			Self::dispatch(app_id.into(), &message)
+			let mut channel = make_inbound_channel::<T>(channel_id);
+			channel.submit(&relayer, &message)
```

Comment:
I'm assuming dropping the call to `T::Verifier::verify` is temporary for the sake of dev. Just a heads up that I'm busy with e2e debugging of message proofs. So it would be good to keep verification around. Consider adding a config param like https://github.com/Snowfork/polkadot-ethereum/blob/aa43d98b0665245b8a7327d17e5b316d0c8928fc/parachain/pallets/verifier-lightclient/src/mock.rs#L114 to disable verification in the pallet in tests

Reply:
Thanks for the heads up. Though for unit tests I'm just disabling the verifier using this technique

bridge/src/mock.rs
```rust
impl Config for Test {
	type Event = TestEvent;
	type Verifier = ();
	type AppETH = ();
	type AppERC20 = ();
	type MessageCommitment = ();
}
```

primitives/cores/sr/lib:
```rust
impl<AccountId> Verifier<AccountId> for () {
	fn verify(sender: AccountId, app_id: AppId, message: &Message) -> DispatchResult {
		Ok(())
	}
	fn verify_bulk(sender: AccountId, messages: &[(AppId, Message)]) -> DispatchResult {
		Ok(())
	}
}
```